src/App.tsx

```diff
@@ -15,36 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  ByLength = 'ByLength',
+  Alphabetically = 'Alphabetically',
+}
+
+function sortGoods(
+  goods: string[],
+  sortType: SortType | null,
+  reverse: boolean,
+): string[] {
+  const sortedGoods = [...goods].sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.Alphabetically:
+        return good1.localeCompare(good2);
+
+      case SortType.ByLength:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return reverse ? sortedGoods.reverse() : sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sort, setSort] = useState<SortType | null>(null);
+  const [isReverse, setIsReverse] = useState<boolean>(false);
+
+  const isResetButtonVisible = sort || isReverse;
+
+  const resultSortedGoods =
+    sort || isReverse
+      ? sortGoods(goodsFromServer, sort, isReverse)
+      : goodsFromServer;
+
+  const handleToggleSortType = (type: SortType) => {
+    setSort(type === sort ? null : type);
+  };
+
+  const handleReset = () => {
+    setIsReverse(false);
+    setSort(null);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sort !== SortType.Alphabetically,
+          })}
+          onClick={() => handleToggleSortType(SortType.Alphabetically)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sort !== SortType.ByLength,
+          })}
+          onClick={() => handleToggleSortType(SortType.ByLength)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !isReverse,
+          })}
+          onClick={() => setIsReverse(!isReverse)}
```

Comment:
```suggestion
          onClick={() => setIsReverse(prev => !prev)}
```

Always use callback when update state based on previous value